### PR TITLE
WFLY-19306 Upgrade to Hibernate ORM 6.6.0.Final / WFLY-19632 Upgrade to Hibernate Search 7.2.0.Final

### PR DESCRIPTION
**Pull request by marko-bekhta**
A draft PR to get the tests running against a combination of these two PRs:
https://github.com/wildfly/wildfly/pull/18129
https://github.com/wildfly/wildfly/pull/18119

That update Hibernate ORM to 6.6.0 and Hibernate Search to 7.2.1

____

<--- THIS SECTION IS AUTOMATICALLY GENERATED BY WILDFLY GITHUB BOT. ANY MANUAL CHANGES WILL BE LOST. --->

> Wildfly issue links:
> * [WFLY-19306](https://issues.redhat.com/browse/WFLY-19306)
> * [WFLY-19633](https://issues.redhat.com/browse/WFLY-19633)
> * [WFLY-19632](https://issues.redhat.com/browse/WFLY-19632)
> * [WFLY-19631](https://issues.redhat.com/browse/WFLY-19631)
> * [WFLY-19630](https://issues.redhat.com/browse/WFLY-19630)

<--- END OF WILDFLY GITHUB BOT REPORT --->

More information about the [wildfly-bot[bot]](https://github.com/wildfly/wildfly-github-bot)

### CODE REVIEW

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/webtxem/TestServlet.java
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/webtxem/TestServlet.java:64]: @marko-bekhta I want to make sure I understand this correctly.

First, calling setXXX for a property annotated with @ GeneratedValue is an anti-pattern. So what we've been doing here is pretty broken.

With ORM 6.6, because the property has @ GeneratedValue, ORM can definitively determine the entity is detached is no DB record with key '1L', so this fails with an OLE.

Right? 

I think the key point is 'what we've been doing here is pretty broken'. This kind of app needs migration but such an app was always flawed.

Comment (bstansberry) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/webtxem/TestServlet.java:64]: Nit, only if you happen to change something for some other reason, and even then it's optional:

Perhaps:

"Previously, we specified the WebJPAEntity id value but as mentioned ^ because the @Id field is annotated with @GeneratedValue we can no longer do that."

I had to parse things a bit to understand this; maybe that would have helped.

No big deal.

**Comment by yrodiere** [comment]
Hey @**Brian Stansberry** @**Scott Marlow** @**Marko Bekhta**, FYI the TCK challenge was accepted: https://github.com/jakartaee/persistence/issues/644
Also, tests passed.
So I think we're good to merge, assuming Brian's comment is not a strong requirement. If it is, we can adapt the comment, but that'll mean another full CI run.

**Comment by yrodiere** [comment]
> <--- THIS SECTION IS AUTOMATICALLY GENERATED BY WILDFLY GITHUB BOT. ANY MANUAL CHANGES WILL BE LOST. --->
> 
> > Wildfly issue links:
> > 
> > * [WFLY-19306](https://issues.redhat.com/browse/WFLY-19306)
> > * [WFLY-19633](https://issues.redhat.com/browse/WFLY-19633)
> > * [WFLY-19632](https://issues.redhat.com/browse/WFLY-19632)
> > * [WFLY-19631](https://issues.redhat.com/browse/WFLY-19631)
> > * [WFLY-19630](https://issues.redhat.com/browse/WFLY-19630)
> 
> <--- END OF WILDFLY GITHUB BOT REPORT --->
> 
> More information about the [wildfly-bot[bot]](https://github.com/wildfly/wildfly-github-bot)

Want to take this opportunty to state that this feature is awesome @xstefank :)

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/webtxem/TestServlet.java
Comment (marko-bekhta) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/webtxem/TestServlet.java:64]: > I think the key point is 'what we've been doing here is pretty broken'. This kind of app needs migration but such an app was always flawed.

yes 👍🏻 
I also opened this follow-up PR https://github.com/wildfly/wildfly/pull/18207 to update the comment (including your suggestion) and keep this one ready for merging without waiting for another CI run.

**Review by scottmarlow — APPROVED** [review]

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @marko-bekhta @yrodiere and @scottmarlow!

**Pull request merged**
